src/main/java/org/candlepin/subscriptions/files/ReportingAccountWhitelist.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.files;
+
+import org.candlepin.subscriptions.ApplicationProperties;
+
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * Loads a list of account numbers that are permitted to access the Tally and Capacity API.
+ */
+@Component
+public class ReportingAccountWhitelist implements ResourceLoaderAware {
+
+    private PerLineFileSource source;
+    private boolean isDevMode;
+
+    public ReportingAccountWhitelist(ApplicationProperties props) {
+        String resourceLocation = props.getReportingAccountWhitelistResourceLocation();
+        source = resourceLocation != null ? new PerLineFileSource(resourceLocation) : null;
+        this.isDevMode = props.isDevMode();
+    }
+
+    public boolean hasAccount(String account) throws IOException {
+        // Whitelist any account when running in dev mode!
+        if (isDevMode) {
+            return true;
+        }
+        return source != null && source.list().contains(account);
```

Comment:
I just realized this will cause the file to be re-read on every request. Probably not problematic due to caching, but it might be worth leaving a note that this is an area we could apply application-level caching if necessary.

Reply:
I was thinking about reading the file only once on creation, but since that would force us to have to restart the app to have the list updated, I didn't bother. I will add a note.

Reply:
DONE